### PR TITLE
Create a new format for testing

### LINKED ISSUE #26: Create a new format to test ASTNodes trees.
**Summary.**
Create a new format to test ASTNodes trees.

### PR DESCRIPTION
**Pull request by johan-smc**
### Summary
I changed the test function and created the test cases in files from different folders, the asserts were made by comparing the input with the same input expected.

### Test Plan
Create the test files, and run that.

### Related Issues
close #26

### CODE REVIEW

File: src/test/java/ast/mix/2.msg [hunk 1]
```diff
@@ -0,0 +1 @@
+1. should have many node, many children, many sibling
```
Comment (DorianMoreno) [inline src/test/java/ast/mix/2.msg]: Spellcheck
```suggestion
1. should have many nodes, many children, many siblings
```

or maybe
```suggestion
1. should have many nodes, many children and many siblings
```

File: src/test/java/ast/sibling/2.out [hunk 2]
```diff
@@ -0,0 +1,3 @@
+A
+    B
+    C
```
Comment (DorianMoreno) [inline src/test/java/ast/sibling/2.out]: Why do some of the tests have an `.out` and others don't? I'm talking about `2.out` in `/sibling` and `2.out` in `/child`.

File: src/main/java/ast/ASTNodes.kt [hunk 3]
```diff
@@ -2,6 +2,12 @@ package ast
 
 interface ASTNode
 
-data class Child(var father: ASTNode, var child: ASTNode) : ASTNode
-data class Node(var name: String) : ASTNode
-data class Sibling(var siblings: List<ASTNode>) : ASTNode
+data class Child(var father: ASTNode, var child: ASTNode) : ASTNode {
+    override fun toString(): String = "$father\n${child.toString().prependIndent("    ")}"
```
Comment (DorianMoreno) [inline src/main/java/ast/ASTNodes.kt]: GJ finding `.prependIndent()`!

File: src/test/java/ast/sibling/2.out (hunk 2 above)
Comment (johan-smc) [reply src/test/java/ast/sibling/2.out]: Because only write .out files when the format is different than the input, for example, these two cases.

Comment (DorianMoreno) [reply src/test/java/ast/sibling/2.out]: Now it makes sense, thank you!

File: src/main/java/ast/ASTNodes.kt (hunk 3 above)
Comment (juanpa097) [inline src/main/java/ast/ASTNodes.kt]: Can we add this `"    "` as a constant at the top? Same for all hard coded  strings in files bellow.

File: src/test/java/ast/ASTNodesKtTest.kt [hunk 4]
```diff
@@ -3,159 +3,95 @@ package ast
...
         val tree = createParserTree(inputAntlrString)
         val visitor = TreepatVisitorImplementation()
-        val root = visitor.visit(tree)
         // act
-        countNodes(root)
+        val root = visitor.visit(tree)
         // assert
-        assertEquals(countNode, 3)
+        assertEquals(root.toString(), output)
         // TODO - sibling should have only one node
```
Comment (juanpa097) [inline src/test/java/ast/ASTNodesKtTest.kt]: I think we then solved this `TODO`, should we delete it?

Comment (johan-smc) [reply src/test/java/ast/ASTNodesKtTest.kt]: No is solved yet.

File: src/main/java/ast/ASTNodes.kt (hunk 3 above)
Comment (johan-smc) [reply src/main/java/ast/ASTNodes.kt]: Complete

**Comment by juanpa097** [comment]
closes #32

**Pull request merged**